### DISCUSSION TITLE
ProxyJump does not behave as ProxyCommand

### DISCUSSION THREAD

**Discussion by IsaacNez** [blinksh/blink#1629] [Support] [answered]
### Checklist

- [ ] Using latest version on the [App Store](http://itunes.apple.com/app/id1156707581)
- [x] Read the [docs](http://blink.sh/docs/)
- [x] Searched for [existing GitHub issues](https://github.com/blinksh/blink/issues)

### Configuration

Blink Shell Version: Testflight v16.0.2.617
iPadOS 16.1

### Describe the bug

Connecting to a remote host using ProxyJump does not work but using ProxyCommand does work. The login is by using password and the usernames through the different bastions is also different.

I experience the same behavior when using `ssh -J proxyhost remotehost`. 

When configured with ProxyCommand, it does work and the verbose output removes the `[]` surrounding the url from the proxyhost.

### Expected behavior

I can login to the remote host using ProxyJump as using ProxyCommand.

### Blink Verbose Output

```
blink> ssh -vvv remotehost
Setting up connection callbacks.
Starting connection to remotehost
ssh_connect: libssh 0.9.3 (c) 2003-2019 Aris Adamantiadis, Andreas Schneider and libssh contributors. Distributed under the LGPL, please refer to COPYING file for information about your rights, using threading threads_pthread
ssh_socket_connect_proxycommand: Executing proxycommand 'ssh -W [url from proxyhost]:22 remotehost'
ssh_socket_connect_proxycommand: ProxyCommand connection pipe: [26,27]
socket_callback_connected: Socket connection callback: 1 (0)
ssh_connect: Socket connecting, now waiting for the callbacks to work
ssh_connect: current state : 2
ssh_socket_exception_callback: Socket exception callback: 1 (0)
ssh_socket_exception_callback: Socket error: disconnected
connError(msg: "Session Exception - Socket error: disconnected")
Starting connection to remoehost
ssh_connect: current state : 9
Error connecting to remotehost. connError(msg: "Socket error: disconnected")
Session deinit
```

**Comment by carloscabanero** [comment, marked as answer]
Thanks a lot for your report. I'm not able to replicate this. As you noticed, ProxyJump is just a translation to ProxyCommand, but devil is in the details here. Maybe what is going on is that when you use ProxyCommand, you are doing that translation yourself so you are passing everything properly but not with ProxyJump. One thing I have noticed:

`'ssh -W [url from proxyhost]:22 remotehost'`

I think this should actually be:
`'ssh -W [url from remotehost]:22 proxyhost'`

First you connect to the Proxy, and instruct it to create a stdio tunnel to remotehost, and remotehost is "Remotehost" for the Proxy. So you have them the flipped. If you send the Host configuration we may be able to help further.

**Reply by IsaacNez** [reply]
Hi @carloscabanero ,

Sorry to get back to you so late. Well, it looks like the problem with the latest updates have been fixed! I can properly use ProxyJump both from config and command line. And it now also works with vscode!

Thank you very much!

**Reply by carloscabanero** [reply]
Glad the latest fixes there worked out for your scenario too!
